proposals/0000-stable-sort.md

```diff
@@ -0,0 +1,78 @@
+# Document Sorting as Stable
+
+* Proposal: [SE-NNNN](0000-stable-sort.md)
+* Author: [Nate Cook](https://github.com/natecook1000)
+* Review Manager: TBD
+* Status: **Awaiting review**
+* Implementation: [apple/swift PR #60936](https://github.com/apple/swift/pull/60936)
+
+## Introduction
+
+Swift's sorting algorithm was changed to be stable before Swift 5, but we've never updated the documentation to provide that guarantee. Let's commit to the sorting algorithm being stable so that people can rely on that behavior.
+
+Swift-evolution thread: [Pitch](https://forums.swift.org/t/pitch-document-sorting-as-stable/59880)
+
+## Motivation
+
+A *stable sort* is a sort that keeps the original relative order for any elements that compare as equal or unordered. For example, given this list of players that are already sorted by last name, a sort by first name preserves the original order of the two players named "Ashley":
+
+```swift
+var roster = [
+   Player(first: "Sam", last: "Coffey"),
+   Player(first: "Ashley", last: "Hatch"),
+   Player(first: "Kristie", last: "Mewis"),
+   Player(first: "Ashley", last: "Sanchez"),
+   Player(first: "Sophia", last: "Smith"),
+]
+
+roster.sort(by: { $0.first < $1.first })
+// roster == [
+//    Player(first: "Ashley", last: "Hatch"),
+//    Player(first: "Ashley", last: "Sanchez"),
+//    Player(first: "Kristie", last: "Mewis"),
+//    Player(first: "Sam", last: "Coffey"),
+//    Player(first: "Sophia", last: "Smith"),
+// ]
+```
+
+For users who are unaware that many sorting algorithms aren't stable, an unstable sort can be surprising. Preserving relative order is an expectation set by software like spreadsheets, where sorting by one column, and then another, is a way to complete a sort based on multiple properties.
+
+Sort stability isn't always observable. When a collection is sorted based on the elements' `Comparable` conformance, like sorting an array of integers, "unordered" elements are typically indistinguishable. In general, sort stability is important when elements are sorted based on a subset of their properties.
+
+The standard library `sort()` has long been stable, but the documentation explicitly [doesn't make this guarantee](https://developer.apple.com/documentation/swift/array/sorted()):
```

Comment:
As the documentation from Apple Developer is subject to change, especially after this proposal being accepted, I would suggest we refer to the comment lines from an existing `apple/swift` release branch (eg. `release/5.6`) or a specific commit hash.

Reply:
Agreed—linking to the doc comment in a fixed commit hash seems like the best way to keep this document stable going forward.